src/Assertion/Validator/Pipes/CheckPublicKeySignature.php

```diff
@@ -27,15 +29,15 @@ class CheckPublicKeySignature
      */
     public function handle(AssertionValidation $validation, Closure $next): mixed
     {
-        $publicKey = openssl_pkey_get_public($validation->credential->public_key);
-
-        if (!$publicKey) {
+        try {
+            $publicKey = openssl_pkey_get_public($validation->credential->public_key);
+        } catch (OpensslException) {
             throw AssertionException::make('Stored Public Key is invalid.');
```

Comment:
Wouldn't it be nicer to add the caught exception to `AssertionException` as the previous exception? This would keep all the information what went wrong until the exception is finally handled.

Of course this also holds for similar places like this one.